pkg/plugin/aggregation/aggregator.go

```diff
@@ -285,24 +285,24 @@ func (a *Aggregator) handleResult(result *plugin.Result) error {
 	}
 
 	// Create the output directory for the result.  Will be of the
-	// form .../plugins/:results_type/:node.json (for DaemonSet plugins) or
+	// form .../plugins/:results_type/:node/filename.json (for DaemonSet plugins) or
 	// .../plugins/:results_type.json (for Job plugins)
-	resultsFile := path.Join(a.OutputDir, result.Path())
-	resultsDir := path.Dir(resultsFile)
+	resultsDir := filepath.Join(a.OutputDir, result.Path())
+	resultFile := filepath.Join(resultsDir, result.Filename)
 
 	if err := os.MkdirAll(resultsDir, 0755); err != nil {
-		errors.Wrapf(err, "couldn't create directory %v", resultsDir)
+		errors.Wrapf(err, "couldn't create directory %q", resultsDir)
```

Comment:
Errors when iterating made it clear that quotes here would be easier due to confusion when the error string ends up there.